test/app-tap/CMakeLists.txt

```diff
@@ -6,3 +6,103 @@ target_link_libraries(module_api msgpuck)
 set(CMAKE_C_FLAGS "-Wall -Wextra -std=c99")
 set(CMAKE_C_FLAGS_DEBUG "-g -O0 -Werror")
 set(CMAKE_C_FLAGS_RELWITHDEBINFO "-O2")
+
+make_lua_path(LUA_PATH
+  PATHS
+    ./?.lua
+    ${CMAKE_CURRENT_SOURCE_DIR}/lua/?.lua
+    # Needed for test-run/test_run.lua.
+    ${PROJECT_SOURCE_DIR}/test-run/?.lua
+)
+
+make_lua_path(LUA_CPATH
+  PATHS
+  ${CMAKE_CURRENT_BINARY_DIR}/?.so
+)
+
+set(TEST_SUITE_NAME "app-tap")
+set(TARANTOOL_BIN $<TARGET_FILE:tarantool>)
+
+message(STATUS "Add test suite ${TEST_SUITE_NAME}")
+
+# XXX: The call produces both test and target <app-tap-deps>
+# as a side effect.
+_add_test_suite_target(${TEST_SUITE_NAME}
+  LABELS ${TEST_SUITE_NAME}
+  DEPENDS module_api
+)
+
+# XXX: Setting PATH is required by app-tap/minimal.test.lua.
+# I would pass TARANTOOL_BIN instead and update the test.
+set(ENV_PATH $ENV{PATH})
+set(ENV_PATH "${PROJECT_BINARY_DIR}/src:${ENV_PATH}")
+list(APPEND TEST_ENV "LUA_PATH=${LUA_PATH}")
+list(APPEND TEST_ENV "LUA_CPATH=${LUA_CPATH}")
+list(APPEND TEST_ENV "PATH=${ENV_PATH}")
+
+file(GLOB tests ${CMAKE_CURRENT_SOURCE_DIR} *.test.lua)
+foreach(test_path ${tests})
+  get_filename_component(test_name ${test_path} NAME)
+  # FIXME: By default, GLOB lists directories.
+  # Directories are omitted in the result if LIST_DIRECTORIES
+  # is set to false. New in version CMake 3.3.
+  if(${test_name} STREQUAL ${TEST_SUITE_NAME})
+    continue()
+  endif()
+  set(test_title "test/${TEST_SUITE_NAME}/${test_name}")
+  set(work_dir "${CMAKE_CURRENT_BINARY_DIR}/${test_name}-workdir")
+  file(MAKE_DIRECTORY ${work_dir})
+  set(TEST_CMD "${RM} -f ${work_dir}/* && ${TARANTOOL_BIN} ${test_path}")
```

Comment:
Side note: May be we can add (instead?) this step to the `POST_BUILD`, see [build_events](https://cmake.org/cmake/help/latest/command/add_custom_command.html#build-events).